pages/moderator/[jam].js

```diff
@@ -0,0 +1,266 @@
+import {
+  Badge,
+  Box,
+  Button,
+  Flex,
+  GridItem,
+  Heading,
+  Spacer,
+  Stack,
+  Switch,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Text,
+} from '@chakra-ui/react';
+import { ArrowBackIcon, ChatIcon, LockIcon } from '@chakra-ui/icons';
+import Layout from 'components/Layout';
+import AdminHeader from 'components/AdminHeader';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import Link from 'next/link';
+import merge from 'lodash.merge';
+import ModeratorNewStatementCard from '../../components/ModeratorNewStatementCard';
+
+const LiveStatementCard = ({ statement, buttonText, onClick }) => {
+  return (
+    <Box
+      border="1px"
+      p={5}
+      borderRadius="md"
+      borderColor="gray.200"
+      my={4}
+      backgroundColor="white"
+    >
+      <Text pb={5}>{statement.text}</Text>
+
+      <Flex>
+        <Box>
+          {statement.isUserSubmitted ? (
+            <Text color="gray.600">
+              <ChatIcon /> Participant submitted{' '}
+              {new Date(
+                statement.createdAt?._seconds * 1000,
+              ).toUTCString()}
+            </Text>
+          ) : (
+            <Text color="gray.600">
+              <LockIcon /> Moderator submitted{' '}
+              {new Date(
+                statement.createdAt?._seconds * 1000,
+              ).toUTCString()}
+            </Text>
+          )}
+        </Box>
+
+        <Spacer />
+
+        <Button variant="outline" onClick={onClick}>
+          {buttonText}
+        </Button>
+      </Flex>
+    </Box>
+  );
+};
+
+const Jam = () => {
+  const router = useRouter();
+  const { jam: jamUrlPath } = router.query;
+  const [jam, setJam] = useState({});
+  const [published, setPublished] = useState();
+  const [location, setLocation] = useState();
+  const [approvedStatements, setApprovedStatements] = useState([]);
+  const [rejectedStatements, setRejectedStatements] = useState([]);
+  const [newStatements, setNewStatements] = useState([]);
+  const [patchSuccess, setPatchTrigger] = useState();
+
+  useEffect(() => {
+    setLocation(window.location.origin);
+  }, []);
+
+  useEffect(() => {
+    if (router.isReady) {
+      loadJam();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router.isReady]);
+
+  useEffect(() => {
+    if (!jam.statements) {
+      return;
+    }
+
+    setApprovedStatements(
+      jam.statements.filter((statement) => statement.state === 1),
+    );
+    setRejectedStatements(
+      jam.statements.filter((statement) => statement.state === -1),
+    );
+    setNewStatements(
+      jam.statements.filter((statement) => statement.state === 0),
+    );
+  }, [jam, patchSuccess]);
```

Comment:
Are you using this `patchSuccess` as a toggle just to trigger this `useEffect`?

Reply:
Yep, I couldn't get it to detect changes on `jam.statements` by including `jam`, or `jam.statements` here, it's a bit of an issue with using an array of objects as an observable: https://stackoverflow.com/questions/63536876/react-hooks-useeffect-not-being-triggered-on-array-of-objects